presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -106,7 +110,7 @@ public synchronized void setResourceOvercommit()
     {
         // Allow the query to use the entire pool. This way the worker will kill the query, if it uses the entire local general pool.
         // The coordinator will kill the query if the cluster runs out of memory.
-        maxMemory = memoryPool.getMaxBytes();
+        maxUserMemory = memoryPool.getMaxBytes();
```

Comment:
`maxTotalMemory` should also be overwritten here. 

Please also update `testResourceOverCommit` to set `query.max-total-memory-per-node`.